### PR TITLE
Upgrade php-cs-fixer to ^2.10 + fix PHP version to 5.6 in composer

### PR DESCRIPTION
**Pull request by Awkan**
# Initial review
- Upgrade php-cs-fixer to native version `^2.10` (instead of Pedrotroller's one)
- Upgrade PHP minimum requirement to 5.6 since this is our minimum requirement yet
- Remove duplicate code in `.travis.yml` file

# 1st review updates
- Update PHP CS Fixer config to `^2.10`
    - (Replace deprecated `align_double_arrow ` ;) )
- Travis
    - [TMP before revision] Remove composer flag `--prefer-lowest` for PHP 5.6
    - Add Branch `2.x` to travis CI
- Update `CHANGELOG.md`
    - Forgotten Change in Dockerfile (previous PR of composer add)
    - Current php-cs-fixer change

# 2nd review updates
- PHP CS Fixer config from `^2.10` to `^2.2` (remove conflictual case with `composer update --prefer-lowest`)

### CODE REVIEW

File: .php_cs.dist [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                'break', 'continue', 'extra', 'return', 'throw', 'use', 'parenthesis_brace_block', 'square_brace_block', 'curly_brace_block'
+            ]
+        ],
+        'no_useless_else' => true,
+        'no_useless_return' => true,
+        'ordered_imports' => true,
+        'phpdoc_order' => true,
+    ])
+    ->setFinder($finder)
+;
```
Comment (Nek-) [inline .php_cs.dist]: Missing new line at end of file.

File: composer.json [hunk 2]
```diff
@@ -37,9 +37,9 @@
     "require-dev": {
         "bossa/phpspec2-expect": "~2.0",
         "leanphp/phpspec-code-coverage": "~3.0",
-        "pedrotroller/php-cs-custom-fixer": "~1.4.0",
         "phpspec/phpspec": "~3.0",
-        "phpspec/prophecy": "~1.6"
+        "phpspec/prophecy": "~1.6",
+        "friendsofphp/php-cs-fixer": "^2.10"
```
Comment (Nek-) [inline composer.json]: 👍

**Comment by Nek-** [comment]
build is broken, we need to investigate

**Comment by Awkan** [comment]
@Nek- The travis bug is due to composer flags. We should speak about them. To make tests pass, I have to remove `--prefer-lowest` flag for PHP 5.6. Composer execution time pass from 10 min + (with timeout) to 2 min ...

**Comment by Awkan** [comment]
@athos7933 @babeou @Nek- @valentin-biig New version set (by considering composer flags "bug" with travis)
- Update PHP CS Fixer config to `^2.10`
    - (Replace deprecated `align_double_arrow ` ;) )
- Update PHP minimum version from `>=^5.4` to `^5.6 || >=7.0`
- Travis
    - [TMP before revision] Remove composer flag `--prefer-lowest` for PHP 5.6
    - Remove duplicate code
    - Add Branch `2.x` to travis CI

**Comment by Awkan** [comment]
Before merging, I have to check why `composer update --prefer-source --prefer-lowest --prefer-stable` fails (to much time to do this command) since adding `friendsofphp/php-cs-fixer`.

**Comment by Nek-** [comment]
@Awkan what's the state of this PR?

**Comment by Awkan** [comment]
Thanks for the ping @Nek- .    
I didn't investigate enough before. Then I've done it today.

# Problem
The problem is caused my something like composer version conflict between dependencies (and dependencies of dependencies). Then composer hangs at "Resolving dependencies through SAT". 

There is 2 solutions that seem's to work locally to resolve this problem:
1 - Restrict bundle usage to SF3 (Remove ~2.8 extension which cause conflict with last `php-cs-fixer` versions)
2 - Downgrade `php-cs-fixer` minimal dependency to `^2.2` (actually at `^2.10` or `^2.11`)

The good alternative seems to be the second one, this will not break compatibility of Symfony.

**Comment by Awkan** [comment]
@Nek- Tests are now passing
A new review can be made (@athos7933 @babeou @valentin-biig you can review again too if you want)


Should I run a csFixer on whole project ?

**Pull request merged**